rfcs/20250430-host-side-scalars/README.md

```diff
@@ -0,0 +1,180 @@
+# Host-side scalars Support in oneDNN Primitives
+
+## Motivation
+
+Currently, primitives assume that all inputs passed at execute time
+are on the same device as the one they execute on (with the exception
+of reorder).
+
+A new request is to pass scalar values that reside on host device
+to GPU kernels at execution time.
+A good example is the scale for SDPA operator in Pytorch [^1], which is
+a python variable on the host and not a torch.tensor object.
```

Comment:
Is this motivation still relevant given that there is a way to pass the scalar to oneDNN? Was there a concern about associated overhead?

I just remembered that we received some complaints in the past about the CPU, where passing a single scalar requires creating the entire memory object, which incurs unnecessary overhead. So maybe it would make sense to look at the issue beyond the GPU.

Reply:
> Is this motivation still relevant given that there is a way to pass the scalar to oneDNN?

The issue here is more about where that scalar resides. If it was a torch.tensor, it would reside on the device and we would be good (aligned with current API). However, because it is on the host, it is not aligned with our current API for execution time arguments (those take only device memory, that live on the same engine as the primitive, except for reorder).

Reply:
hey @densamoilov, I've updated the RFC, let me know if this thread is still relevant (or please resolve otherwise)

Reply:
Thank you, its looks good to me.